src/colorizer/utils/url_utils.ts

```diff
@@ -53,40 +60,58 @@ export function fetchWithTimeout(
+ * - `track`: integer track number.
+ * - `time`: integer frame number.
+ * - `thresholds`: array of feature threshold.
+ * - `range`: array of two numbers, representing the min and max of the color map range.
  *
  * @returns
  * - If no parameters are present or valid, returns an empty string.
  * - Else, returns a string of URL parameters that can be appended to the URL directly (ex: `?collection=<some_url>&time=23`).
  */
-export function stateToUrlParamString(state: Partial<UrlParams>): string {
+export function stateToUrlQueryString(state: Partial<UrlParams>): string {
   // arguments as more data gets stored to the URL.
 
   // Get parameters, ignoring null/empty values
   const includedParameters: string[] = [];
-  const { collection, dataset, feature, track, time } = state;
-
-  // Don't include collection parameter in URL if it matches the default.
-  if (
-    collection &&
-    collection !== DEFAULT_COLLECTION_PATH &&
-    collection !== DEFAULT_COLLECTION_PATH + "/" + DEFAULT_COLLECTION_FILENAME
-  ) {
-    includedParameters.push(`${URL_PARAM_COLLECTION}=${encodeURIComponent(collection)}`);
+
+  if (state.collection) {
+    includedParameters.push(`${URL_PARAM_COLLECTION}=${encodeURIComponent(state.collection)}`);
+  }
+  if (state.dataset) {
+    includedParameters.push(`${URL_PARAM_DATASET}=${encodeURIComponent(state.dataset)}`);
   }
-  if (dataset) {
-    includedParameters.push(`${URL_PARAM_DATASET}=${encodeURIComponent(dataset)}`);
+  if (state.feature) {
+    includedParameters.push(`${URL_PARAM_FEATURE}=${encodeURIComponent(state.feature)}`);
   }
-  if (feature) {
-    includedParameters.push(`${URL_PARAM_FEATURE}=${encodeURIComponent(feature)}`);
+  if (state.track !== undefined) {
+    includedParameters.push(`${URL_PARAM_TRACK}=${state.track}`);
   }
-  if (track && track >= 0) {
-    includedParameters.push(`${URL_PARAM_TRACK}=${track}`);
+  if (state.time !== undefined) {
+    includedParameters.push(`${URL_PARAM_TIME}=${state.time}`);
   }
-  if (time && time > 0) {
-    // time = 0 is ignored because it's the default frame.
-    includedParameters.push(`${URL_PARAM_TIME}=${time}`);
+  if (state.thresholds && state.thresholds.length > 0) {
+    // Thresholds are saved as a comma-separated list of `featureName:unit:min:max`
+    // featureName is encoded in case it contains special characters (":" or ",")
+    // TODO: Are there better characters I can be using here? ":" and "," take up
+    // more space in the URL.
+    const thresholdsString = state.thresholds
+      .map((threshold) => {
+        const featureName = encodeURIComponent(threshold.featureName);
+        // Note that THRESHOLD_UNIT_UNDEFINED (="!") is not encoded here, so that it won't conflict
+        // with normal feature units names. (Users should not be using "!" as a unit anyway, but just in case.)
+        const featureUnit =
+          threshold.unit !== undefined ? encodeURIComponent(threshold.unit) : THRESHOLD_UNIT_UNDEFINED;
```

Comment:
Would empty string have worked for THRESHOLD_UNIT_UNDEFINED?  `name::min:max` instead of `name:!:min:max` ?  Just wondering.
Also what if threshold.unit === "" ? Then that is what you'd get anyway.

Reply:
Currently empty strings are considered different from a feature unit being `undefined`? `threshold.unit` is either `undefined` or a string type.

If `THRESHOLD_UNIT_UNDEFINED = ""`, unit values of `undefined` and `""` would BOTH map to the same value, which would make the URL saving/parsing lossy. For now, `undefined` and `""` values mean different things in the interface, though I guess there could be a discussion about whether we should allow `""` as a valid unit type anyways.

Let me know if that makes sense?

Reply:
The real question is: what's the difference between undefined and ""?  Don't they both mean we have a unitless feature?

Reply:
Hmm, okay, I started digging into the code for unit handling again. From what I see there isn't a situation where `undefined` is handled differently from `""` other than in tests. Does it make sense to just replace `undefined` values with an empty string on Dataset load, then?

Reply:
I'm going to just make that change and update the relevant tests, I think.

Reply:
My hope was that it would be simpler to handle in that way.